### PR TITLE
para optimize 

### PR DESCRIPTION
**Pull request by scf0220**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: libs/cosmos-sdk/baseapp/baseapp_parallel.go [hunk 1]
```diff
@@ -149,22 +148,14 @@ func (app *BaseApp) calGroup() {
...
 
 	if txSize == 0 {
 		return make([]*abci.ResponseDeliverTx, 0)
 	}
-	pm.init()
+
+	pm := app.parallelTxManage
+	pm.init(txs, app.deliverState.ctx.BlockHeight(), app.deliverState.ms)
 
 	app.getExtraDataByTxs(txs)
```
Comment (lyh169) [inline libs/cosmos-sdk/baseapp/baseapp_parallel.go:179]: getExtraDataByTxs inside can according to the cpu core start go routine

File: libs/cosmos-sdk/baseapp/baseapp_parallel.go [hunk 2]
```diff
@@ -410,411 +389,366 @@ func newExecuteResult(r abci.ResponseDeliverTx, ms sdk.CacheMultiStore, counter
...
 
 func newParallelTxManager() *parallelTxManager {
 	isAsync := sm.DeliverTxsExecMode(viper.GetInt(sm.FlagDeliverTxsExecMode)) == sm.DeliverTxsExecModeParallel
-	return &parallelTxManager{
+	para := &parallelTxManager{
 		blockGasMeterMu:  sync.Mutex{},
 		isAsyncDeliverTx: isAsync,
-		workgroup:        newAsyncWorkGroup(),
+		resultCh:         make(chan int, maxTxNumberInParallelChan),
+		stop:             make(chan struct{}, 1),
```
Comment (lyh169) [inline libs/cosmos-sdk/baseapp/baseapp_parallel.go:595]: `resultCh:         make(chan int, maxTxNumberInParallelChan)` maxTxNumberInParallelChan should have actual tx num, prevent the std tx large than maxTxNumberInParallelChan in one block

File: libs/cosmos-sdk/baseapp/baseapp_parallel.go (hunk 1 above)
Comment (scf0220) [reply libs/cosmos-sdk/baseapp/baseapp_parallel.go:179]: done

File: libs/cosmos-sdk/baseapp/baseapp_parallel.go (hunk 2 above)
Comment (scf0220) [reply libs/cosmos-sdk/baseapp/baseapp_parallel.go:595]: done

**Review by lyh169 — DISMISSED** [review]

File: libs/cosmos-sdk/baseapp/baseapp_synclist.go [hunk 3]
```diff
@@ -2,10 +2,61 @@ package baseapp
...
 )
 
+type cacheRWSetList struct {
+	mtx sync.Mutex
+	mps *list.List
+}
+
+func newCacheRWSetList() *cacheRWSetList {
+	return &cacheRWSetList{
+		mtx: sync.Mutex{},
```
Comment (BananaLF) [inline libs/cosmos-sdk/baseapp/baseapp_synclist.go]: mutex need not to init

File: libs/cosmos-sdk/baseapp/baseapp_synclist.go [hunk 4]
```diff
@@ -2,10 +2,61 @@ package baseapp
...
 import (
 	"container/list"
-	"github.com/okex/exchain/libs/cosmos-sdk/store/types"
 	"sync"
+
+	"github.com/okex/exchain/libs/cosmos-sdk/store/types"
 )
 
+type cacheRWSetList struct {
+	mtx sync.Mutex
```
Comment (BananaLF) [inline libs/cosmos-sdk/baseapp/baseapp_synclist.go:11]: suggest use rw mutex

File: libs/cosmos-sdk/baseapp/baseapp_synclist.go [hunk 5]
```diff
@@ -2,10 +2,61 @@ package baseapp
...
+	return c.mps.Len()
+}
+
+func (c *cacheRWSetList) PutRwSet(rw types.MsRWSet) {
+	c.mtx.Lock()
+	defer c.mtx.Unlock()
+	c.mps.PushBack(rw)
+}
+
+func (c *cacheRWSetList) GetRWSet() types.MsRWSet {
```
Comment (BananaLF) [inline libs/cosmos-sdk/baseapp/baseapp_synclist.go:33]: use pointer much better

**Review by cwbhhjl — DISMISSED** [review]

**Review by BananaLF — APPROVED** [review]

**Pull request merged**